ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/AbstractTool.java

```diff
@@ -0,0 +1,77 @@
+package org.opensearch.ml.engine.tools;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.opensearch.ml.common.spi.tools.Parser;
+import org.opensearch.ml.common.spi.tools.Tool;
+
+import java.util.Map;
+
+public abstract class AbstractTool implements Tool {
```

Comment:
It's better move this class to spi module, so that tools in plugin own repository still can re-use it

Reply:
Good suggestion, done!